src/Compilers/CSharp/Test/Emit2/Attributes/AttributeTests.cs

```diff
@@ -10799,6 +10799,74 @@ void verify(ModuleSymbol module)
             }
         }
 
+        [Fact, WorkItem(58837, "https://github.com/dotnet/roslyn/issues/58837")]
+        public void GenericAttribute_NestedType()
+        {
+            var source = @"
+using System;
+
+class Container<T>
+{
+    [Attr<T>]
+    class C1
+    {
+    }
+    
+    [AttrContainer<T>.Attr]
+    class C2
+    {
+    }
+    
+    [AttrContainer<T>.B.Attr]
+    class C3
+    {
+    }
+}
+
+class Attr<T> : Attribute { }
+
+class AttrContainer<T>
+{  
+    public class Attr : Attribute
+    {
+        public T Value { get; set; }
+    }
+    
+    public class B
+    {
+        public class Attr : Attribute
+        {
+        }
+    }
+}
+
+[A<(int A, int B)>.B]
+class C
+{
+}
+
+class A<T>
+{  
+    public class B : Attribute
```

Comment:
>B [](http://example.com/codeflow?start=17&length=1)

Please add a test for `[A<int A, int B)>.B]` where the inner type is named `BAttribute`.

Reply:
2c71c355975227a42c17e03acbcd856acccf626e